### PR TITLE
v5 migration guide

### PR DESCRIPTION
**Pull request by BPScott**
### WHY are these changes introduced?

Docs

### WHAT is this pull request doing?

Docs

### CODE REVIEW

**Comment by BPScott** [comment]
Heya folks, I've took a stab at interpreting the affect of breaking changes per https://github.com/Shopify/polaris-react/blob/master/UNRELEASED.md and reviewing the changes in https://github.com/Shopify/polaris-react/pull/2364. If you changed something I'd appreciate checking that my interpretation is right and I've not missed any nuance.

File: UNRELEASED.md [hunk 1]
```diff
@@ -11,7 +11,12 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 - Removed `ResourceList.FilterControl` component. The `FilterControl` component is available under a private name for legacy Shopify applications, but it should not be relied upon and might be deleted at any point ([#2047](https://github.com/Shopify/polaris-react/pull/2047) [#3116](https://github.com/Shopify/polaris-react/pull/3116))
 - Removed `AppBridge`, `ResourcePicker` and `Loading`, `Modal`, `Page`, `Toast` App Bridge render delegation ([#2046](https://github.com/Shopify/polaris-react/pull/2046))
 - Dropped support for iOS 9 ([#2195](https://github.com/Shopify/polaris-react/pull/2195))
-- Moved styles from `global.scss` to `AppProvider`. This change only affects applications using the `esnext` build (applications importing `@shopify/polaris/styles.css` aren’t affected), who no longer need to import the `@shopify/polaris/esnext/global.scss` file. An empty `global.scss` was kept in, to ensure applications using sewing-kit \<v0.113.0 still build ([#2392](https://github.com/Shopify/polaris-react/pull/2392))
+- Radically reworked our built output.
```
Comment (alex-page) [inline UNRELEASED.md:14]: I don't get a lot of understanding from "Radically reworked our built output." What do you think about this?

```suggestion
- Simplified the built output by putting necessary files in a dist folder and removing unnecessary files.
```

File: documentation/guides/migrating-from-v4-to-v5.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
+# Migrating from v4 to v5
+
+Polaris v5.0.0 ([full release notes](https://github.com/Shopify/polaris-react/releases/tag/v5.0.0)) removes most features deprecated in v4.x.x releases. This file describes all code updates required to stay up to date.
+
+## AppBridge integrations removed
+
+In v3 and v4 `AppProvider` could accept `apiKey` `shopOrigin` and `forceRedirect` props that would configure AppBridge delegation in our `ResourcePicker`, `Loading`, `Modal`, `Page` and `Toast` components.
```
Comment (alex-page) [inline documentation/guides/migrating-from-v4-to-v5.md:7]: > In v3 and v4 `AppProvider` could accept `apiKey` `shopOrigin` and `forceRedirect` props that would configure AppBridge delegation in our `ResourcePicker`, `Loading`, `Modal`, `Page` and `Toast` components.

I think we should rephrase this to say that in v5 these props are removed. I know the title above says integrations removed but it's not clear if these are the integrations that are removed or what works in v3 and v4.

File: documentation/guides/migrating-from-v4-to-v5.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+  return (
+    <AppProvider apiKey={config.apiKey} shopOrigin={config.shopOrigin}>
+      <Modal /* ... *//>
+    </AppProvider>
+  );
+}
+
+
+// NEw
```
Comment (alex-page) [inline documentation/guides/migrating-from-v4-to-v5.md:28]:
```suggestion
// New
```

File: documentation/guides/migrating-from-v4-to-v5.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+      </Provider>
+    <AppProvider>
+  );
+}
+```
+
+## AppProvider's i18n translations array is now preferred language first
+
+AppProvider's i18n prop accepts an array of translation dictionaries so that if
+a translation key in one language is not found it can try a fallback language.
```
Comment (alex-page) [inline documentation/guides/migrating-from-v4-to-v5.md:48]:
```suggestion
AppProvider's i18n prop accepts an array of translation dictionaries so that if a translation key in one language is not found it can try a fallback language.
```

File: documentation/guides/migrating-from-v4-to-v5.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+Our scss entrypoint has been removed, and you should use the above styles file instead. If you require our mixins/functions you should also import the our public-api entrypoint.
+
+```diff
+
+- @import 'path_to_node_modules/@shopify/polaris/styles';
++ @import 'path_to_node_modules/@shopify/polaris/dist/styles.css';
++ /* Optional - only needed if you use our mixins/functions */
++ @import 'path_to_node_modules/@shopify/polaris/dist/styles/public-api';
+```
```
Comment (alex-page) [inline documentation/guides/migrating-from-v4-to-v5.md:102]:
```suggestion
```diff
- @import 'path_to_node_modules/@shopify/polaris/styles';
+ @import 'path_to_node_modules/@shopify/polaris/dist/styles.css';
+ /* Optional - only needed if you use our mixins/functions */
+ @import 'path_to_node_modules/@shopify/polaris/dist/styles/public-api';
```
```

File: documentation/guides/migrating-from-v4-to-v5.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+```diff
+- @import 'path_to_node_modules/@shopify/polaris/styles';
++ @import 'path_to_node_modules/@shopify/polaris/dist/styles.css';
++ /* Optional - only needed if you use our mixins/functions */
++ @import 'path_to_node_modules/@shopify/polaris/dist/styles/public-api';
+
+## Build output changes - @shopify/sewing-kit integrations only
+
+Apps using `@shopify/sewing-kit` - Shopify's in-house opinionated build toolkit - will need to update to at least v0.132.2 to make use of the reworked "esnext" build output in polaris v5.
```
Comment (alex-page) [inline documentation/guides/migrating-from-v4-to-v5.md:104]: I don't think "Shopify's in-house opinionated build toolkit" adds value here.

```suggestion
Apps using `@shopify/sewing-kit` will need to update to at least v0.132.2 to make use of the reworked "esnext" build output in polaris v5.
```

Comment (BPScott) [reply documentation/guides/migrating-from-v4-to-v5.md:105]: I was trying to enforce the idea that this doesn't apply to partners, and thus they can skip over this section. By adding this little bit I was trying to preempt thoughts of "what's sewing-kit? does this apply to me?" that may occur we just refer to sewing-kit by name.

Do you think just referring to the package by name is enough for partners to know this doesn't apply to them, is there there some other way we can help them reach that "cool, this doesn't apply to me" realization quickly?

File: documentation/guides/migrating-from-v4-to-v5.md (hunk 5 above)
Comment (BPScott) [reply documentation/guides/migrating-from-v4-to-v5.md]: Pretty sure you didn't want want to get rid of that final ``` that closes the  fenced code block - that might be nested codeblocks in markdown being an ass I think that's bit me before. I've added it back :)

File: documentation/guides/migrating-from-v4-to-v5.md (hunk 6 above)
Comment (alex-page) [reply documentation/guides/migrating-from-v4-to-v5.md:105]: I think referring to the package is enough. Happy for other opinions though.

File: UNRELEASED.md [hunk 7]
```diff
@@ -11,7 +11,12 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
...
+  - `styles.css` has moved to `dist/styles.css`
+  - `styles.min.css has been removed - import`dist/styles.css` instead
+  - `styles.scss has been removed - import`dist/styles.css`and`dist/styles/\_public-api.scss` instead
```
Comment (kaelig) [inline UNRELEASED.md:17]:
```suggestion
  - `styles.scss` has been removed - import `dist/styles.css` and `dist/styles/\_public-api.scss` instead
```

File: UNRELEASED.md [hunk 8]
```diff
@@ -11,7 +11,12 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 - Removed `ResourceList.FilterControl` component. The `FilterControl` component is available under a private name for legacy Shopify applications, but it should not be relied upon and might be deleted at any point ([#2047](https://github.com/Shopify/polaris-react/pull/2047) [#3116](https://github.com/Shopify/polaris-react/pull/3116))
 - Removed `AppBridge`, `ResourcePicker` and `Loading`, `Modal`, `Page`, `Toast` App Bridge render delegation ([#2046](https://github.com/Shopify/polaris-react/pull/2046))
 - Dropped support for iOS 9 ([#2195](https://github.com/Shopify/polaris-react/pull/2195))
-- Moved styles from `global.scss` to `AppProvider`. This change only affects applications using the `esnext` build (applications importing `@shopify/polaris/styles.css` aren’t affected), who no longer need to import the `@shopify/polaris/esnext/global.scss` file. An empty `global.scss` was kept in, to ensure applications using sewing-kit \<v0.113.0 still build ([#2392](https://github.com/Shopify/polaris-react/pull/2392))
+- Radically reworked our built output.
+  - `styles.css` has moved to `dist/styles.css`
+  - `styles.min.css has been removed - import`dist/styles.css` instead
```
Comment (kaelig) [inline UNRELEASED.md:16]:
```suggestion
  - `styles.min.css` has been removed - import `dist/styles.css` instead (the tiny performance hit should be unnoticeable by users)
```

File: documentation/guides/migrating-from-v4-to-v5.md [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+
+## Dependencies
+
+The peer dependencies on `react` and `react-dom` have been increased to 16.9.0, to allow us to use react's Profiler component in the future. Use the package manager of your choice to install a recent version of these pacakges.
+
+Apps built using `@shopify/sewing-kit` - Shopify's in-house opinionated build toolkit - will need to upgrade to at least version `0.132.2`.
+
+## CDN usage
+
+As a result of build output changes, the location of our static stylesheet for CSS-only usage has been moved. Update any stylesheet imports to use the new path:
```
Comment (kaelig) [inline documentation/guides/migrating-from-v4-to-v5.md:173]: I try to avoid "our", in case it's perceived as "not yours".

```suggestion
As a result of build output changes, the location of the static stylesheet for CSS-only usage has been moved. Update any stylesheet imports to use the new path:
```

File: documentation/guides/migrating-from-v4-to-v5.md [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+
+### Removed Exports
+
+#### NavigationMessageProps
+
+The `NavigationMessageProps` type has been removed, as Navigation's Message internal subcomponent has been removed.
+
+## Dependencies
+
+The peer dependencies on `react` and `react-dom` have been increased to 16.9.0, to allow us to use react's Profiler component in the future. Use the package manager of your choice to install a recent version of these pacakges.
```
Comment (kaelig) [inline documentation/guides/migrating-from-v4-to-v5.md]: For consistency

```suggestion
The peer dependencies on `react` and `react-dom` have been increased to `16.9.0`, opening the door to using the [React Profiler](https://reactjs.org/blog/2018/09/10/introducing-the-react-profiler.html) in the future. Use the package manager of your choice to install a recent version of these packages.
```

File: documentation/guides/migrating-from-v4-to-v5.md [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+## Build output changes - general consumers
+
+Our compiled styles file has moved. Imports must be updated:
+
+```diff
+- import '@shopify/polaris/styles.css';
++ import '@shopify/polaris/dist/styles.css';
+```
+
+Our minified styles file has been removed, and you should use the above file instead.
```
Comment (kaelig) [inline documentation/guides/migrating-from-v4-to-v5.md:108]:
```suggestion
The minified styles file has been removed, and you should use the above file instead.
```

File: documentation/guides/migrating-from-v4-to-v5.md [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+  return (
+-    <AppProvider i18n={i18n.translations.reverse()}>{/* App content */}</AppProvider>
++    <AppProvider i18n={i18n.translations}>{/* App content */}</AppProvider>
+  );
+}
+```
+
+## Build output changes - general consumers
+
+Our compiled styles file has moved. Imports must be updated:
```
Comment (kaelig) [inline documentation/guides/migrating-from-v4-to-v5.md:101]:
```suggestion
The compiled styles file has moved. Imports must be updated:
```

File: documentation/guides/migrating-from-v4-to-v5.md (hunk 6 above)
Comment (kaelig) [reply documentation/guides/migrating-from-v4-to-v5.md]: I'm with Ben on this one, unless you feel strongly about removing this part of the sentence 👍

File: documentation/guides/migrating-from-v4-to-v5.md [hunk 13]
```diff
@@ -0,0 +1,156 @@
...
+
+Apps using `@shopify/sewing-kit` - Shopify's in-house opinionated build toolkit - will need to update to at least v0.132.2 to make use of the reworked "esnext" build output in polaris v5.
+
+Apps using `@shopify/sewing-kit` and Polaris v4 also had to have an explicit import of `@shopify/polaris/esnext/styles/global.scss` in the same file as they imported and used `AppProvider`. This is no longer required, this import can be removed. The styles in this file will be implicitly imported when you use `AppProvider`.
+
+```diff
+- import '@shopify/polaris/esnext/styles/global.scss';
+```
+
+Sewing-kit apps can use our sass public API by configuring their sass plugin's autoInclude option. The path to the sass API file has now changed and should be updated accordingly:
```
Comment (kaelig) [inline documentation/guides/migrating-from-v4-to-v5.md:134]:
```suggestion
Sewing-kit apps can use Polaris React’s Sass public API by configuring their Sass plugin's `autoInclude` option. The path to the Sass API file has now changed and should be updated accordingly:
```

File: documentation/guides/migrating-from-v4-to-v5.md [hunk 14]
```diff
@@ -0,0 +1,156 @@
...
+
+### EmptyState
+
+The `EmptyState` component's `centeredLayout` prop has been removed as the illustration's location is now handled automatically by opting into the new design language. You should remove any usage of this prop.
+
+### Removed Exports
+
+#### NavigationMessageProps
+
+The `NavigationMessageProps` type has been removed, as Navigation's Message internal subcomponent has been removed.
```
Comment (kaelig) [inline documentation/guides/migrating-from-v4-to-v5.md:163]:
```suggestion
The `NavigationMessageProps` type has been removed, as `Navigation`'s `Message` internal subcomponent has been removed.
```

File: documentation/guides/migrating-from-v4-to-v5.md [hunk 15]
```diff
@@ -0,0 +1,181 @@
...
+
+```diff
+import enTranslations from '@shopify/polaris/locales/en.json';
+import frTranslations from '@shopify/polaris/locales/fr.json';
+
+function App() {
+  return (
+    // French translations as the primary language, english as a fallback
+-    <AppProvider i18n={[enTranslations, frTranslations]}>
++    <AppProvider i18n={frTranslations, enTranslations]}>
```
Comment (kaelig) [inline documentation/guides/migrating-from-v4-to-v5.md:66]:
```suggestion
+    <AppProvider i18n={[frTranslations, enTranslations]}>
```

File: documentation/guides/migrating-from-v4-to-v5.md [hunk 16]
```diff
@@ -0,0 +1,181 @@
...
+```
+
+Our minified styles file has been removed, and you should use the above file instead.
+
+```diff
+- import '@shopify/polaris/styles.min.css';
++ import '@shopify/polaris/dist/styles.css';
+```
+
+Our scss entrypoint has been removed, and you should use the above styles file instead. If you require our mixins/functions you should also import the our public-api entrypoint.
```
Comment (kaelig) [inline documentation/guides/migrating-from-v4-to-v5.md:115]:
```suggestion
The scss entrypoint has been removed, and you should use the above `styles.css` file instead. If you require Polaris React’s Sass mixins and functions, you should also import the `public-api` entrypoint.
```

File: documentation/guides/migrating-from-v4-to-v5.md [hunk 17]
```diff
@@ -0,0 +1,181 @@
...
+```diff
+- @import 'path_to_node_modules/@shopify/polaris/styles';
++ @import 'path_to_node_modules/@shopify/polaris/dist/styles.css';
++ /* Optional - only needed if you use our mixins/functions */
++ @import 'path_to_node_modules/@shopify/polaris/dist/styles/public-api';
+```
+
+## Build output changes - @shopify/sewing-kit integrations only
+
+Apps built using `@shopify/sewing-kit` - Shopify's in-house opinionated build toolkit - will need to update to at least v0.132.2 to make use of the reworked "esnext" build output in polaris v5.
```
Comment (kaelig) [inline documentation/guides/migrating-from-v4-to-v5.md:126]:
```suggestion
Apps built using `@shopify/sewing-kit` - Shopify's in-house opinionated build toolkit - will need to update to at least v0.132.2 to make use of the reworked "esnext" build output in Polaris React v5.
```

File: UNRELEASED.md (hunk 8 above)
Comment (BPScott) [reply UNRELEASED.md:16]: I actually made some tweaks to how styles.css gets generated and now the diff between an old styles.min.css and a new styles.css is 0.7kb post gzip :D

File: UNRELEASED.md (hunk 1 above)
Comment (BPScott) [reply UNRELEASED.md]: Have reworded this differently: "Moved several of our build artifacts into a `dist` folder:"

File: documentation/guides/migrating-from-v4-to-v5.md [hunk 18]
```diff
@@ -0,0 +1,181 @@
...
+In v3 and v4 `AppProvider` could accept `apiKey` `shopOrigin` and `forceRedirect` props that would configure AppBridge delegation in our `ResourcePicker`, `Loading`, `Modal`, `Page` and `Toast` components. This functionality has been removed.
+
+In v5 AppBridge integration has been removed from these components:
+
+- The `AppProvider` component's `apiKey`, `shopOrigin` and `forceRedirect` have been removed.
+- The `Modal` component's `size` and `message` props have been removed as they only had meaning within an AppBridge app.
+- The `Page` component's `forceRender` prop has been removed as it only had meaning within an AppBridge app.
+- The `ResourcePicker` component has been removed as it only had meaning within an AppBridge app.
+
+AppBridge behavour has been abstracted into a separate `@shopify/app-bridge-react` package. If you require the AppBridge implementation of these components then use the components from the [`@shopify/app-bridge-react` package](https://shopify.dev/tools/app-bridge/react-components).
```
Comment (dleroux) [inline documentation/guides/migrating-from-v4-to-v5.md:16]:
```suggestion
AppBridge behavior has been abstracted into a separate `@shopify/app-bridge-react` package. If you require the AppBridge implementation of these components then use the components from the [`@shopify/app-bridge-react` package](https://shopify.dev/tools/app-bridge/react-components).
```

File: documentation/guides/migrating-from-v4-to-v5.md [hunk 19]
```diff
@@ -0,0 +1,181 @@
...
+    <AppProvider>
+      <Provider config={config}>
+        <Modal /* ... *//>
+      </Provider>
+    <AppProvider>
+  );
+}
+```
+
+## AppProvider's i18n translations array is now preferred language first
```
Comment (AndrewMusgrave) [inline documentation/guides/migrating-from-v4-to-v5.md:50]:
```suggestion
## AppProvider's i18n translations array is now the preferred language first
```

File: documentation/guides/migrating-from-v4-to-v5.md [hunk 20]
```diff
@@ -0,0 +1,181 @@
...
+```
+
+The minified styles file has been removed, and you should use the above file instead.
+
+```diff
+- import '@shopify/polaris/styles.min.css';
++ import '@shopify/polaris/dist/styles.css';
+```
+
+The scss entrypoint has been removed, and you should use the above `styles.css` file instead. If you require Polaris React’s Sass mixins and functions, you should also import the `public-api` entrypoint.
```
Comment (AndrewMusgrave) [inline documentation/guides/migrating-from-v4-to-v5.md:115]:
```suggestion
The scss entry point has been removed, and you should use the above `styles.css` file instead. If you require Polaris React’s Sass mixins and functions, you should also import the `public-api` entry point.
```

File: documentation/guides/migrating-from-v4-to-v5.md [hunk 21]
```diff
@@ -0,0 +1,181 @@
...
+
+The `ResourceList.FilterControl` subcomponent has been removed. You should use the `Filters` component to provide filtering options.
+
+### Page
+
+The `Page` component's `singleColumn` prop has been removed. You should replace any usage of this prop with the `narrowWidth` prop which has identical functionality.
+
+### EmptyState
+
+The `EmptyState` component's `centeredLayout` prop has been removed as the illustration's location is now handled automatically by opting into the new design language. You should remove any usage of this prop.
```
Comment (AndrewMusgrave) [inline documentation/guides/migrating-from-v4-to-v5.md:157]:
```suggestion
The `EmptyState` component's `centeredLayout` prop has been removed as the illustration's location is now handled automatically by opting in to the new design language. You should remove any usage of this prop.
```

File: UNRELEASED.md [hunk 22]
```diff
@@ -11,7 +11,12 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 - Removed `ResourceList.FilterControl` component. The `FilterControl` component is available under a private name for legacy Shopify applications, but it should not be relied upon and might be deleted at any point ([#2047](https://github.com/Shopify/polaris-react/pull/2047) [#3116](https://github.com/Shopify/polaris-react/pull/3116))
 - Removed `AppBridge`, `ResourcePicker` and `Loading`, `Modal`, `Page`, `Toast` App Bridge render delegation ([#2046](https://github.com/Shopify/polaris-react/pull/2046))
 - Dropped support for iOS 9 ([#2195](https://github.com/Shopify/polaris-react/pull/2195))
-- Moved styles from `global.scss` to `AppProvider`. This change only affects applications using the `esnext` build (applications importing `@shopify/polaris/styles.css` aren’t affected), who no longer need to import the `@shopify/polaris/esnext/global.scss` file. An empty `global.scss` was kept in, to ensure applications using sewing-kit \<v0.113.0 still build ([#2392](https://github.com/Shopify/polaris-react/pull/2392))
+- Moved several of our build artifacts into a `dist` folder:
```
Comment (AndrewMusgrave) [inline UNRELEASED.md:14]: Do we need a PR link for the main bullet point?

File: documentation/guides/migrating-from-v4-to-v5.md [hunk 23]
```diff
@@ -0,0 +1,181 @@
...
+    <AppProvider apiKey={config.apiKey} shopOrigin={config.shopOrigin}>
+      <Modal /* ... *//>
+    </AppProvider>
+  );
+}
+
+
+// New
+import {AppProvider} from '@shopify/polaris';
+import {Provider, TitleBar} from '@shopify/app-bridge-react';
```
Comment (AndrewMusgrave) [inline documentation/guides/migrating-from-v4-to-v5.md:35]:
```suggestion
import {Provider, Modal} from '@shopify/app-bridge-react';
```

File: documentation/guides/migrating-from-v4-to-v5.md (hunk 19 above)
Comment (BPScott) [inline documentation/guides/migrating-from-v4-to-v5.md:50]:
```suggestion
## AppProvider's i18n translations array is now ordered by preferred language first
```

File: UNRELEASED.md (hunk 22 above)
Comment (BPScott) [reply UNRELEASED.md]: Added

File: documentation/guides/migrating-from-v4-to-v5.md [hunk 24]
```diff
@@ -0,0 +1,224 @@
...
+
+Polaris v5.0.0 ([full release notes](https://github.com/Shopify/polaris-react/releases/tag/v5.0.0)) removes most features deprecated in v4.x.x releases. This file describes all code updates required to stay up to date.
+
+## AppBridge integrations removed
+
+In v3 and v4 `AppProvider` could accept `apiKey` `shopOrigin` and `forceRedirect` props that would configure AppBridge delegation in our `ResourcePicker`, `Loading`, `Modal`, `Page` and `Toast` components. This functionality has been removed.
+
+In v5 AppBridge integration has been removed from these components:
+
+- The `AppProvider` component's `apiKey`, `shopOrigin` and `forceRedirect` have been removed.
```
Comment (kyledurand) [inline documentation/guides/migrating-from-v4-to-v5.md:11]:
```suggestion
- The `AppProvider` component’s `apiKey`, `shopOrigin` and `forceRedirect` have been removed.
```
I'm not sure this is important in markdown files but I try to use curly quotes for public facing docs
